sdk/scheduler/src/main/java/com/mesosphere/sdk/http/EndpointUtils.java

```diff
@@ -74,35 +73,36 @@ public static String toAutoIpEndpoint(
      * Returns the correct DNS hostname for accessing the Scheduler API given the provided service name.
      */
     public static String toSchedulerAutoIpHostname(String serviceName, SchedulerConfig schedulerConfig) {
-        return toAutoIpHostname("marathon", serviceName, schedulerConfig);
+        return toAutoIpHostname(schedulerConfig.getMarathonName(), serviceName, schedulerConfig);
     }
 
     /**
      * Returns the correct DNS hostname:port endpoint for accessing the Scheduler API given the provided service name.
      */
     public static String toSchedulerAutoIpEndpoint(String serviceName, SchedulerConfig schedulerConfig) {
-        return toAutoIpEndpoint("marathon", serviceName, schedulerConfig.getApiServerPort(), schedulerConfig);
+        return toAutoIpEndpoint(
```

Comment:
Quick question. Why doesn't this call `toSchedulerAutoIpHostname` internally?

Reply:
Looks like the dependency chain is:
```
toAutoIpDomain
    toAutoIpHostname
        toSchedulerAutoIpHostname
        toAutoIpEndpoint
            toSchedulerAutoIpEndpoint
```